### PR TITLE
VDS-719: Add error report to import processing blade

### PR DESCRIPTION
**Pull request by ilyawzrd**

### CODE REVIEW

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataImporter.cs [hunk 1]
```diff
@@ -96,6 +139,22 @@ public async Task ImportAsync(ImportDataRequest request, Action<ImportProgressIn
                         var validationResult = await _importProductPricesValidator.ValidateAsync(importProductPrices, ruleSet: request.ImportMode.ToString());
 
                         var invalidImportProductPrices = validationResult.Errors.Select(x => (x.CustomState as ImportValidationState)?.InvalidImportProductPrice).Distinct().ToArray();
+
+                        var errorsInfos = validationResult.Errors.Select(x => new { Message = x.ErrorMessage, ImportProductPrice = (x.CustomState as ImportValidationState)?.InvalidImportProductPrice }).ToArray();
+
+                        var errorsGroups = errorsInfos.GroupBy(x => x.ImportProductPrice);
+
+                        foreach (var g in errorsGroups)
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataImporter.cs]: please use meaningful variable name

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs [hunk 2]
```diff
@@ -83,30 +84,45 @@ public async Task<bool> FetchAsync()
                 return false;
             }
 
-            // CSV Reader can only move forward, i.e. Skip will not work: after reading N items we can read only next N items
-            var records = _csvReader.GetRecords<CsvPrice>().Take(PageSize).ToArray();
-            CurrentPageNumber++;
+            var recordTuples = new List<(CsvPrice, string, int)>();
+
+            for (var i = 0; i < PageSize && _csvReader.Read(); i++)
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs]: There is async csv helper read method. Why we don't use it?

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataImporter.cs (hunk 1 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataImporter.cs]: done

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs (hunk 2 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs]: switched to ReadAsync

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs [hunk 3]
```diff
@@ -41,6 +43,34 @@ public CsvPagedPriceDataSource(IProductSearchService productSearchService, Strea
 
         public int PageSize { get; }
 
+
+        public string GetHeaderRaw()
+        {
+            string result = null;
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs]: maybe var result = string.Empty?

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs [hunk 4]
```diff
@@ -24,6 +24,8 @@ public sealed class CsvPagedPriceDataSource : ICsvPagedPriceDataSource
         private readonly CsvReader _csvReader;
         private int? _totalCount;
 
+
+
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs]: Remove empty lines

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs [hunk 5]
```diff
@@ -41,6 +41,33 @@ public CsvPagedPriceDataSource(IProductSearchService productSearchService, Strea
 
         public int PageSize { get; }
 
+        public string GetHeaderRaw()
+        {
+            string result = null;
+
+            var streamPosition = _stream.Position;
+            _stream.Seek(0, SeekOrigin.Begin);
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPagedPriceDataSource.cs:49]: If GetHeaderRaw would be called between FetchAsync calls, does streams would be rewinded?

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPriceImportReporter.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+
+        private string GetLine(ImportError importError)
+        {
+            var result = $"{importError.Error}{_configuration.Delimiter}{importError.RawRow.TrimEnd()}";
+
+            return result;
+        }
+
+        public void WriteHeader(string header)
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPriceImportReporter.cs]: Public methods should be first, then protected and private are the last

File: src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPriceImportReporter.cs [hunk 7]
```diff
@@ -28,22 +28,21 @@ public void Write(ImportError error)
...
+        {
+            _streamWriter.WriteLine($"{ErrorsColumnName}{_configuration.Delimiter}{header}");
+        }
+
         public void Dispose()
         {
             _streamWriter.Dispose();
         }
 
-
```
Comment (pushnitsa) [inline src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPriceImportReporter.cs]: private methods should be separated by two empty lines.
https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md#formatting

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**